src/tests/src/libs/antares/study/short-term-storage-input/short-term-storage-input.cpp

```diff
@@ -0,0 +1,317 @@
+#define BOOST_TEST_MODULE "test short term storage"
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/included/unit_test.hpp>
+#include <yuni/io/file.h>
+#include <filesystem>
+#include <random>
+#include <fstream>
+
+#include <study.h>
+
+#include "series.h"
+
+#define SEP Yuni::IO::Separator
+
+using namespace std;
+using namespace Antares::Data;
+
+std::string getFolder()
+{
+    std::filesystem::path tmpDir = std::filesystem::temp_directory_path();
+    return tmpDir.string();
+}
+
+void resizeFillVectors(ShortTermStorage::Series& series, double value, unsigned int size)
+{
+    series.maxInjectionModulation.resize(size, value);
+    series.maxWithdrawalModulation.resize(size, value);
+    series.inflows.resize(size, value);
+    series.lowerRuleCurve.resize(size, value);
+    series.upperRuleCurve.resize(size, value);
+}
+
+void createIndividualFileSeries(const std::string& path, double value, unsigned int size)
+{
+    std::ofstream outfile(path);
+
+    for (unsigned int i = 0; i < size; i++)
+        outfile << value << std::endl;
+
+    outfile.close();
+}
+
+void createIndividualFileSeries(const std::string& path, unsigned int size)
+{
+    std::default_random_engine rnd{std::random_device{}()};
+    std::uniform_real_distribution<double> dist(0, 1);
+
+    std::ofstream outfile;
+    outfile.open(path, std::ofstream::out | std::ofstream::trunc);
+
+    for (unsigned int i = 0; i < size; i++)
+        outfile << dist(rnd) << std::endl;
+
+    outfile.close();
+}
+
+void createFileSeries(double value, unsigned int size)
+{
+    std::string folder = getFolder();
+
+    createIndividualFileSeries(folder + SEP + "PMAX-injection.txt", value, size);
+    createIndividualFileSeries(folder + SEP + "PMAX-withdrawal.txt", value, size);
+    createIndividualFileSeries(folder + SEP + "inflows.txt", value, size);
+    createIndividualFileSeries(folder + SEP + "lower-rule-curve.txt", value, size);
+    createIndividualFileSeries(folder + SEP + "upper-rule-curve.txt", value, size);
+}
+
+void createFileSeries(unsigned int size)
+{
+    std::string folder = getFolder();
+
+    createIndividualFileSeries(folder + SEP + "PMAX-injection.txt", size);
+    createIndividualFileSeries(folder + SEP + "PMAX-withdrawal.txt", size);
+    createIndividualFileSeries(folder + SEP + "inflows.txt", size);
+    createIndividualFileSeries(folder + SEP + "lower-rule-curve.txt", size);
+    createIndividualFileSeries(folder + SEP + "upper-rule-curve.txt", size);
+}
```

Comment:
I'd rather not have random generation in unit tests. It can fluctuate `ie` give different results in a different run/platform.

Reply:
It can fluctuate but should always return a value between 0 and 1 no ? This test is used to check if loading a file with different values work, using random seems like a good idea but i can do something like **line(n) = 0.0001 * n** to fill the vector